corehq/apps/accounting/interface.py

```diff
@@ -566,9 +567,9 @@ def _invoice_to_row(invoice):
                 contact_info.state_province_region,
                 contact_info.postal_code,
                 contact_info.country,
-                invoice.date_start,
-                invoice.date_end,
-                invoice.date_due,
+                format_datatables_data(invoice.date_start, invoice.date_start),
+                format_datatables_data(invoice.date_end, invoice.date_end),
+                format_datatables_data(invoice.date_due if invoice.date_due else "None", invoice.date_due),
```

Comment:
I think these changes are likely unnecessary. If you look at [format_datatables_data](https://github.com/dimagi/commcare-hq/blob/4bc8de017615b7af70cd4f9726fb0fde4886f3f2/corehq/apps/reports/util.py#L252-L258), this code shouldn't really do anything. We're turning 'Oct 1, 2020' into {'html': 'Oct 1, 2020', 'sort_key': 'Oct 1, 2020'}. As far as I can tell, the intent of this function is to support having different display and sort values; here, we're using the same values, so its redundant. 

The issue, however, comes from a bug elsewhere in the code. These values are used in [tabular_cell.html](https://github.com/dimagi/commcare-hq/blob/4bc8de017615b7af70cd4f9726fb0fde4886f3f2/corehq/apps/reports/templates/reports/async/partials/tabular_cell.html#L1-L4). From the code, it looks like the intent is that the title will either be the sort key or the base value, and the text will either be the html attribute or the base value. So, `format_datatables_data` _should_ be optional. The bug here is:
```python
if col.sort_key != ""
```

For some reason, this template is explicitly checking against an empty string, rather than whether or not it has a value. Because Django template 'if' conditionals [handle all truthiness](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#if), this conditional should just be changed to:
```python
if col.sort_key
```

Once that is done, we can change:
`invoice.date_due if invoice.date_due else "None"` to just `invoice.date_due` -- `date_due` just gets rendered as text, so `None` gets rendered as `"None"` anyway.

Reply:
Hi Matt, I wonder how did you trace to `tabular_cell.html` ? After making the changes you suggested, `date_start`  and `date_end` is sorting properly, but `date_due` is not anymore. Date due will only sort the last two rows now.
![image](https://github.com/dimagi/commcare-hq/assets/39149002/84221a26-1d5b-4fb5-8d80-765896f9fe5b)

Actually, I don't have a very clear mind of why my changes work. I just follow the patterns I noticed in our codebase. If `format_datatables_data` didn't really do anything, why after I remove it, the sorting broken again?

Reply:
Fixed in 54893f9.